File: bot-service/src/core/use_cases/get_expenses.py
from src.core.domain.expenses import Expenses
from src.core.domain.user import UsersRepository


class GetExpenses:
    def __init__(self, users_repository: UsersRepository):
        self._users_repository = users_repository

    def execute(self, telegram_id: str) -> list[Expenses]:
        user = self._users_repository.get(telegram_id)
        return user.expenses

File: connector-service/src/core/use_cases/GetExpenses.ts
import {IExpensesService} from "../domain/expenses/IExpensesService";
import {TelegramId} from "../domain/TelegramId";
import {Expenses} from "../domain/expenses/Expenses";
import {format} from "./Utils";
import {ResponseChat} from "./ResponseChat";

export class GetExpenses {
    constructor(private readonly expensesService: IExpensesService) {
    }

    async execute(telegramId: TelegramId): Promise<ResponseChat> {
        return this.toResponseChat(await this.expensesService.get(telegramId));
    }

    toResponseChat(expenses: Expenses[]): ResponseChat {
        if (expenses.length == 0) {
            return {
                message: `No expenses charged yet`
            }
        }
        
        const message = expenses.map(expense => {
            return `📅 Added at: ${expense.addedAt}\n💵 Amount: $${expense.amount}\n🧾 Category: ${format(expense.category)}\n✅ Description: ${expense.description}\n`;
        }).join('\n');

        return {message: message};
    }

}